src/include/tcop/tcop.h

```diff
@@ -73,8 +77,38 @@ class TrafficCop {
   // InitBindPrepStmt - Prepare and bind a query from a query string
   std::shared_ptr<Statement> PrepareStatement(const std::string &statement_name,
                                               const std::string &query_string,
+                                              std::string &error_message,
+                                              const size_t thread_id = 0);
+
+  // PortalExec - Execute query string
+  ResultType ExecuteStatementJDBC(const std::string &query,
+                              std::vector<StatementResult> &result,
+                              std::vector<FieldInfo> &tuple_descriptor,
+                              int &rows_changed, std::string &error_message,
+                              const size_t thread_id = 0);
+  ResultType ExecuteStatementJDBC(
+      const std::shared_ptr<Statement> &statement,
+      const std::vector<type::Value> &params, const bool unnamed,
+      std::shared_ptr<stats::QueryMetric::QueryParams> param_stats,
+      const std::vector<int> &result_format,
+      std::vector<StatementResult> &result, int &rows_change,
+      std::string &error_message, const size_t thread_id = 0);
+
+  // ExecutePrepStmt - Helper to handle txn-specifics for the plan-tree of a
+  // statement
+  executor::ExecuteResult ExecuteStatementPlanJDBC(
+      const planner::AbstractPlan *plan, const std::vector<type::Value> &params,
+      std::vector<StatementResult> &result,
+      const std::vector<int> &result_format, const size_t thread_id = 0);
+
+  // InitBindPrepStmt - Prepare and bind a query from a query string
+  std::shared_ptr<Statement> PrepareStatementJDBC(const std::string &statement_name,
+                                              const std::string &query_string,
                                               std::string &error_message);
 
+
+
+
```

Comment:
That's a good question. I hacked code based on psql and people told me jdbc has solved transaction handle. So I separate some methods in traffic cop.